# HG changeset patch
# User Rik <rik@octave.org>
# Date 1344355262 25200
#      Tue Aug 07 09:01:02 2012 -0700
# Node ID 4e0ab3b2841899bcda5379763bf3ffd579cd5746
# Parent  ca222e6932b7db1f85f81d8deeb3faded7844fa8
Eliminate duplicate PTHREAD_CFLAGS inclusion in build process.

* configure.ac: Add PTHREAD_CFLAGS to CXXFLAGS.

* build-aux/common.mx: Remove PTHREAD_CFLAGS from AM_CPFLAGS and AM_CXXFLAGS.

diff --git a/build-aux/common.mk b/build-aux/common.mk
--- a/build-aux/common.mk
+++ b/build-aux/common.mk
@@ -112,17 +112,17 @@ UGLY_DEFS = @UGLY_DEFS@
 
 CC = @CC@
 CC_VERSION = @CC_VERSION@
 CONFIGURE_CFLAGS = @CFLAGS@
 CPICFLAG = @CPICFLAG@
 XTRA_CFLAGS = @XTRA_CFLAGS@
 WARN_CFLAGS = @WARN_CFLAGS@
 AM_CFLAGS = $(CONFIGURE_CFLAGS) \
-  $(INCFLAGS) $(XTRA_CFLAGS) $(WARN_CFLAGS) $(PTHREAD_CFLAGS)
+  $(INCFLAGS) $(XTRA_CFLAGS) $(WARN_CFLAGS)
 BUG_CFLAGS = $(XTRA_CFLAGS) $(WARN_CFLAGS) $(CFLAGS)
 
 BUILD_CC = @BUILD_CC@
 BUILD_CFLAGS = @BUILD_CFLAGS@
 
 DEPEND_FLAGS = @DEPEND_FLAGS@
 DEPEND_EXTRA_SED_PATTERN = @DEPEND_EXTRA_SED_PATTERN@
 INCLUDE_DEPS = @INCLUDE_DEPS@
@@ -135,17 +135,17 @@ GRAPHICS_CFLAGS = @GRAPHICS_CFLAGS@
 CXX = @CXX@
 CXX_VERSION = @CXX_VERSION@
 CXXCPP = @CXXCPP@
 CONFIGURE_CXXFLAGS = @CXXFLAGS@
 CXXPICFLAG = @CXXPICFLAG@
 XTRA_CXXFLAGS = @XTRA_CXXFLAGS@
 WARN_CXXFLAGS = @WARN_CXXFLAGS@
 AM_CXXFLAGS = $(CONFIGURE_CXXFLAGS) \
-  $(INCFLAGS) $(XTRA_CXXFLAGS) $(WARN_CXXFLAGS) $(PTHREAD_CFLAGS)
+  $(INCFLAGS) $(XTRA_CXXFLAGS) $(WARN_CXXFLAGS)
 BUG_CXXFLAGS = $(XTRA_CXXFLAGS) $(WARN_CXXFLAGS) $(CXXFLAGS)
 
 BUILD_CXX = @BUILD_CXX@
 BUILD_CXXFLAGS = @BUILD_CXXFLAGS@
 
 NO_UNDEFINED_LDFLAG = @NO_UNDEFINED_LDFLAG@
 
 LD_CXX = @LD_CXX@
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -340,16 +340,17 @@ AC_SUBST(DEPEND_EXTRA_SED_PATTERN)
 
 ### Check for pthread library
 
 AX_PTHREAD
 ## Include pthread libs and flags here in case other tests need them.
 ## They seem to be required for the OpenGL tests on Debian systems.
 LIBS="$PTHREAD_LIBS $LIBS"
 CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
+CXXFLAGS="$CXXFLAGS $PTHREAD_CFLAGS"
 
 ### When compiling math for x87, problems may arise in some code comparing
 ### floating-point intermediate results.
 ### Generally, it helps to store the result in a local volatile variable,
 ### but it also degrades performance.
 ### Thus, we provide a FLOAT_TRUNCATE macro that may be defined to "volatile"
 ### when compiling for x87 target, or left empty for modern SSE math, that
 ### doesn't suffer from this problem at all.
